src/pages/workspaces/migration/WorkspaceItem.ts

```diff
@@ -0,0 +1,115 @@
+import { div, h, span } from 'react-hyperscript-helpers';
+import { icon } from 'src/components/icons';
+import { InfoBox } from 'src/components/PopupTrigger';
+import colors from 'src/libs/colors';
+import * as Utils from 'src/libs/utils';
+import { WorkspaceMigrationInfo } from 'src/pages/workspaces/migration/migration-utils';
+
+export const WorkspaceItem = (workspaceInfo: WorkspaceMigrationInfo) => {
+  const renderMigrationState = () => {
+    const getTransferProgress = (transferType, processed, total) => {
+      if (total === 0) {
+        return `${transferType} Bucket Transfer`;
+      }
+      return `${transferType} Transfer in Progress (${Utils.formatBytes(processed)}/${Utils.formatBytes(total)})`;
+    };
+
+    const text = Utils.cond(
+      [
+        workspaceInfo.outcome === 'failure',
+        () =>
+          span({ style: { color: colors.danger() } }, [
+            'Migration Failed',
+            h(
+              InfoBox,
+              {
+                style: { marginLeft: '0.5rem' },
+                side: 'bottom',
+                tooltip: 'Failure information',
+                size: 18,
+                iconOverride: undefined,
+              },
+              [workspaceInfo.failureReason]
+            ),
+          ]),
+      ],
+      [workspaceInfo.outcome === 'success', () => span(['Migration Complete'])],
+      [workspaceInfo.migrationStep === 'ScheduledForMigration', () => span(['Starting Migration'])],
+      [workspaceInfo.migrationStep === 'PreparingTransferToTempBucket', () => span(['Preparing Original Bucket'])],
+      [
+        workspaceInfo.migrationStep === 'TransferringToTempBucket',
+        () =>
+          span([
+            getTransferProgress(
+              'Initial',
+              workspaceInfo.tempBucketTransferProgress?.bytesTransferred,
+              workspaceInfo.tempBucketTransferProgress?.totalBytesToTransfer
+            ),
+          ]),
+      ],
+      [workspaceInfo.migrationStep === 'PreparingTransferToFinalBucket', () => span(['Creating Destination Bucket'])],
+      [
+        workspaceInfo.migrationStep === 'TransferringToFinalBucket',
+        () =>
+          span([
+            getTransferProgress(
+              'Final',
+              workspaceInfo.finalBucketTransferProgress?.bytesTransferred,
+              workspaceInfo.finalBucketTransferProgress?.totalBytesToTransfer
+            ),
+          ]),
+      ],
+      // If workspace.outcome === 'success', we end earlier with a "Migration Complete" message.
+      // Therefor we shouldn't encounter 'Finished' here, but handling it in case `outcome` updates later.
+      [
+        workspaceInfo.migrationStep === 'FinishingUp' || workspaceInfo.migrationStep === 'Finished',
+        () => span(['Finishing Migration']),
+      ]
+    );
+    const statusIcon = Utils.cond(
```

Comment:
Instead of using separate variables within the same component, it would improve readability if you break them out into their own component (even if they remain in the same file).

For example, this would be something like:
```typescript
interface MigrationStatusIconProps {
  workspaceInfo: WorkspaceMigrationInfo
}
const MigrationStatusIcon = (props: MigrationStatusIconProps): ReactNode => {
  const { workspaceInfo } = props
  const [ iconShape, iconColor]: [IconId, string] = cond(
    [workspaceInfo.outcome === 'failure', () => ['warning-standard', colors.danger()]],
    [workspaceInfo.outcome === 'success', () => ['check', colors. success()]],
   // other cases  ... 
  )

  return div({ style: { display: 'flex' } }, [
      icon(iconShape, {
            size: 18,
            style: { color: iconColor) },
          }),
      div({ style: { display: 'flex', paddingLeft: '0.5rem', alignItems: 'center' } }, [text]),
    ]);
}
```

Reply:
I split them up into their own render methods to improve readability somewhat, but don't really want to make separate components. For one thing, the rendering does differ by type (for instance, with the icon method one of them rotates, and in the next PR one is a different size).